### ISSUE TITLE
ShouldShowToolbarButton() override on FlyoutPage not working

### ISSUE THREAD

**Issue by GODston** [dotnet/maui#15111]
### Description

Im trying to have a custom navigation bar.
I have a FlyoutPage that sets a custom Detail (a NavigationPage that contains a TabbedPage). In the tabbed page constructor i load a custom view to my navigationPage title view, wich works correctly, but since im using a FlyoutPage the burger 'Open Flyout' button is still present.
After looking around i found that i could override the ShouldShowToolbarButton() in my FlyoutPage to make it always return false, thus making it not show the button, but this does not work. Im doing my testing on Android.

### Steps to Reproduce

1- Create new .net MAUI app.
2- Create new FlyoutPage
3- Load a custom ContentPage as Flyout in the constructor (example 1)
4- Load a custom TabbedPage inside a NavigationPage as Detail (emaple 1)
```
<--- Example 1 --->
public Flyout()
{
InitializeComponent();

Flyout = new MainPage();
Detail = new NavigationPage( new TabbedP() );
}
```
5- Load a custom View as TitleView in the TabbedPage
`NavigationPage.SetTitleView(this, new NewContent2());`
6- Override the `ShouldShowToolbarButton` method in the FlyoutPage so it always returns false.
```
public override bool ShouldShowToolbarButton()
{
    return false;
}
```

### Link to public reproduction project repository

https://github.com/GODston/Public

### Version with bug

8.0 previews

### Last version that worked well

Unknown/Other

### Affected platforms

Android, I was *not* able test on other platforms

### Affected platform versions

Android 13 API 33

**Comment by GODston** [comment]
If there is a known workaround to remove the burger button, i would like to know.
Thanks!

**Comment by thomaskaelin** [comment]
While migrating an existing app from Xamarin.Forms to MAUI, I came across this bug as well. I was able to on Android and iOS:

* Android: not working. The method `ShouldShowToolbarButton` is never called. Tested on Android 11, 12 and 13.
* iOS: working. Tested on iOS 16.4.

No workaround known to me.

**Comment by bradencohen** [comment]
Found a pretty ugly workaround to hide the icon in the handler: 

```csharp
public class ToolbarHandler : Microsoft.Maui.Handlers.ToolbarHandler
{
    protected override void ConnectHandler( MaterialToolbar platformView )
    {
        ( VirtualView as Toolbar ).PropertyChanged += ToolbarHandler_PropertyChanged;
        base.ConnectHandler( platformView );
    }

    protected override void DisconnectHandler( MaterialToolbar platformView )
    {
        ( VirtualView as Toolbar ).PropertyChanged -= ToolbarHandler_PropertyChanged;
        base.DisconnectHandler( platformView );
    }

    private void ToolbarHandler_PropertyChanged( object sender, System.ComponentModel.PropertyChangedEventArgs e )
    {
        if( sender is not Toolbar toolbar || !toolbar.DrawerToggleVisible )
        {
            return;
        }

        // This property doesn't invoke a property changed.
        // Somewhere in the source, this is being set to 'true'
        // and doesn't reference ShouldShowToolbarButton().
        // There might be somewhere better to call this but this gets the job done.
        toolbar.DrawerToggleVisible = false;
    }
}
```

I couldn't find anywhere better to call this, and this seems to catch it in my use case. It isn't the best solution because you're assuming that there is a property change after `DrawerToggleVisible` has been updated that will trigger the event.... But that's why its a workaround :P. Don't forget to register the handler in your app builder.

**Comment by XamlTest** [comment]
Verified this on Visual Studio Enterprise 17.8.0 Preview 2.0. Repro on Android 13.0-API33(8.0.0-rc.1), not repro on iOS 16.4 with below Project:
[ExampleFlyoutBurgerButton.zip](https://github.com/dotnet/maui/files/12644417/ExampleFlyoutBurgerButton.zip)

![image](https://github.com/dotnet/maui/assets/97647558/e7690407-fbea-4f39-9319-c02e81deede7)

**Comment by schinkenwuerfel** [comment]
@samhouts I just tested if this is fixed in version 8.0.14 but it seems the issue still is there.
Im overriding the method my FlyoutPage with `public override bool ShouldShowToolbarButton() => false;` but i still have the burger menu button in the titlebar. 
![image](https://github.com/dotnet/maui/assets/45284660/bc162e63-49e3-43f8-8f01-34fac61baacb)

**Comment by bradencohen** [comment]
Seems to be working on main @schinkenwuerfel :

```csharp
class App : Application
{
	protected override Window CreateWindow(IActivationState? activationState)
	{
		// To test shell scenarios, change this to true
		bool useShell = false;

		if (!useShell)
		{
			var flyout = new Flyout();
			flyout.Flyout = new ContentPage { Title = "Flyout" };
			flyout.Detail = new NavigationPage(new MainPage());

			return new Window(flyout);
		}
		else
		{
			return new Window(new SandboxShell());
		}
	}

	class Flyout : FlyoutPage 
	{
		public override bool ShouldShowToolbarButton()
		{
			return false;	
		}
	}
}
```
![image](https://github.com/dotnet/maui/assets/11482832/b54a25bb-e798-4c59-b245-86cbedb883ac)

I have a hunch that the fix made only fixes when the root page is a NavigationPage.
